stream_alert/threat_intel_downloader/threat_stream.py

```diff
@@ -0,0 +1,260 @@
+"""
+Copyright 2017-present, Airbnb Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+   http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+"""
+from datetime import datetime, timedelta
+import json
+
+import boto3
+from botocore.exceptions import ClientError
+import requests
+
+from stream_alert.threat_intel_downloader import LOGGER
+from stream_alert.threat_intel_downloader.exceptions import ThreatStreamCredsError
+
+class ThreatStream(object):
+    """Class to retrieve IOCs from ThreatStream.com and store them in DynamoDB"""
+    _API_URL = 'https://api.threatstream.com'
+    _API_RESOURCE = 'intelligence'
+    _IOC_STATUS = 'active'
+    # max IOC objects received from one API call, default is 0 (equal to 1000)
+    _API_MAX_LIMIT = 1000
+    _API_MAX_INDEX = 1000000
+    _PARAMETER_NAME = 'threat_intel_downloader_api_creds'
+    _MAX_RETRY = 3
+
+    def __init__(self, config):
+        self.ioc_types = config['ioc_types']
+        self.ioc_sources = config['ioc_filters']
+        self.threshold = self._API_MAX_INDEX - self._API_MAX_LIMIT
+        self.region = config['region']
+        self.ioc_keys = config['ioc_keys']
+        self.api_user = None
+        self.api_key = None
+        self._get_api_creds()
+        self.table_name = config['function_name']
+
+    def _get_api_creds(self):
+        """Retrieve ThreatStream API credentials from Parameter Store"""
+        try:
+            ssm = boto3.client('ssm', self.region)
+            response = ssm.get_parameters(
+                Names=[self._PARAMETER_NAME], WithDecryption=True
+            )
+            for cred in response['Parameters']:
+                if cred['Name'] == self._PARAMETER_NAME:
+                    try:
+                        decoded_creds = json.loads(cred['Value'])
+                        self.api_user = decoded_creds['api_user']
```

Comment:
again - only put the items that may result in a `ValueError` in the try block. (move the `self.api_user = decoded_creds['api_user']` and `self.api_key = decoded_creds['api_key']` outside of this)